src/platform/GenericOTARequestorDriver.cpp

```diff
@@ -46,29 +57,86 @@ uint16_t GenericOTARequestorDriver::GetMaxDownloadBlockSize()
     return 1024;
 }
 
+void StartDelayTimerHandler(System::Layer * systemLayer, void * appState)
+{
+    // Cancel the default providers timer, ConnectToProvider() will immediately move
+    // us out of the kIdle state
+    //static_cast<OTARequestor *>(appState)->mOtaRequestorDriver->StopDefaultProvidersTimer();
+
+    static_cast<OTARequestorInterface *>(appState)->SendQuery();
```

Comment:
Trying to follow this and a bit hard to digest. So when a periodic query timer fires, this is the sequence of events with your PR:
-> Driver::StartDelayTimerHandler
  -> Requestor::SendQuery (sets state to kQuerying)
    -> Driver::DriverSendQuery (stops all internal timers in the driver)
     -> Requestor::ConnectToProvider (actually attempts to establish CASE session)

Why the need to go back and forth? Why not just cancel what you need prior to calling `SendQuery` and that function would continue on to calling `ConnectToProvider` as it does right now (or at least previously done in `TriggerImmediateQuery`)?

Reply:
Originally this was a way to make sure that the Requestor moves out of the kIdle state. But you are right, this is needlessly complicated, I fixed it.